lib/collection/src/collection.rs

```diff
@@ -1494,6 +1495,65 @@ impl Collection {
             })
             .collect()
     }
+
+    pub async fn suggest_shard_replica_changes(
+        &self,
+        new_repl_factor: NonZeroU32,
+        all_peers: HashSet<PeerId>,
+    ) -> CollectionResult<Option<HashSet<replica_set::Change>>> {
+        let repl_factor = self.config.read().await.params.replication_factor;
+        match new_repl_factor.cmp(&repl_factor) {
+            std::cmp::Ordering::Less => {
+                // Remove replicas
+                let remove_n = repl_factor.get() - new_repl_factor.get();
+                let shard_holder = self.shards_holder.read().await;
+                let changes: HashSet<_> = shard_holder
+                    .get_shards()
+                    .map(|(shard_id, shard)| match shard {
+                        Shard::ReplicaSet(shard) => Ok((shard_id, shard)),
+                        _ => Err(CollectionError::service_error(
+                            "Shards should all be `ReplicaSet` in this collection".to_string(),
+                        )),
+                    })
+                    .map_ok(|(shard_id, shard)| {
+                        shard
+                            .peer_ids()
+                            .into_iter()
+                            .take(remove_n as usize)
+                            .map(|peer_id| replica_set::Change::Remove(*shard_id, peer_id))
+                            .collect_vec()
+                    })
+                    .flatten_ok()
+                    .try_collect()?;
+                Ok(Some(changes))
+            }
+            std::cmp::Ordering::Equal => Ok(None),
+            std::cmp::Ordering::Greater => {
+                // Add replicas
+                let add_n = new_repl_factor.get() - repl_factor.get();
+                let shard_holder = self.shards_holder.read().await;
+                let changes: HashSet<_> = shard_holder
+                    .get_shards()
+                    .map(|(shard_id, shard)| match shard {
+                        Shard::ReplicaSet(shard) => Ok((shard_id, shard)),
+                        _ => Err(CollectionError::service_error(
+                            "Shards should all be `ReplicaSet` in this collection".to_string(),
```

Comment:
If the original replication factor was 1, doesn't it mean the shard will not be a ReplicaSet?

Reply:
Yeah, makes sense. I will fix it